### ISSUE TITLE
Disable swipe on elements with a specific className

### ISSUE THREAD

**Issue by slevy85** [darenju/react-flip-page#87]
Hi,

Is it possible to disable swipe on specific elements based on there className ?

My problematic is that I have images I need to drag and drop in a `FlipPage` child, when I drag it, it swipes also the page.
I think I can use `disableSwipe` property  when I am dragging, but I think that there could be an easier solution.
What do you think ?

**Comment by slevy85** [comment]
Hi,

I have tried it and it is not working, I suppose because my element has many classes.
I think this means the element must have only class that matches one of the swipe Immune :
```javascript
    const { tagName, className } = e.target;
    const { swipeImmune } = this.props;
    const cancel = swipeImmune.some(testedClassName => className === testedClassName);
```
Can we make it in the reverse way ? Swipe Immune is one string, and we test if it is in one of the classes of the element ?
Or maybe it is two arrays, and we must check if any of one is in the other. 
What do you think ?

**Comment by darenju** [comment]
Oh, sorry, my condition is not good enough. It should be `className.indexOf(testedClassName) !== -1`. I’m fixing it.

**Comment by slevy85** [comment]
Great thank you very much

**Comment by darenju** [comment]
@slevy85 Can you try `react-flip-page@1.5.1` please? The issue should be fixed.

**Comment by slevy85** [comment]
Yes thank you it is working but there is another problem, the dragging works bad, it is like the mouse event is not correctly send to the package I use (react-draggable).

It is strange that when the swipe takes effect the drag works correctly, and when the event is cancelled it doesn't work.
As a note, when the swipe is disabled completely, the dragging works good.

I have no idea why, I will try to look at it

**Comment by darenju** [comment]
Is there any other impact on this?

**Comment by slevy85** [comment]
I don't think so, test suite is ok.

**Comment by slevy85** [comment]
no I am wrong testsuite is not ok

**Comment by slevy85** [comment]
The test 'should stop if event is on a button or a link' fails

```javascript
      const ret = wrapper.instance().startMoving(event);

      expect(ret).toEqual(undefined);
      expect(event.preventDefault).not.toHaveBeenCalled();
```
Maybe we should test weither the state has been modified or not here, what do you think ?

**Comment by darenju** [comment]
I think what you wrote would prevent the links and buttons to be clicked. That’s why the test checks if `event.preventDefault()` was not called. Otherwise, the buttons could not be clicked. Maybe `doNotMove` (shortcut for `event.preventDefault()`) should be called only in the `swipeImmune` case? But what if a button has a `className` that matches the `swipeImmune` prop… ?

**Comment by slevy85** [comment]
Yes I understand what you mean I didn't think of this.
I think you are right we can call donotMove only if it is a `swipeImmune` class.
For the buttons and link it would be ok because the `return` would me made before.
Something like this :
```javascript
if (tagName === 'BUTTON' || tagName === 'A' ) {
      return;
    }
if (cancel) {
      doNotMove(e);
      return;
    }
```

**Comment by darenju** [comment]
Yes, I think that could solve it. I’m releasing this.

**Comment by darenju** [comment]
It’s now available under `react-flip-page@1.5.2`.

**Comment by slevy85** [comment]
Thank you it is working great.
I have tried to verify that the `onClick` event were still working and it does, even on an element with `swipeImmune` class.
I think it is `onMouseDown` and `onTouchStart` that would be cancelled.

**Comment by darenju** [comment]
If I’m not mistaken, cancelling both those event would prevent the click to happen. It’s better not to interfere with the event flow of a click, I think…

**Comment by slevy85** [comment]
Yes it is better not to interfere you are right

Le lun. 10 déc. 2018 à 15:13, Julien Fradin <notifications@github.com> a
écrit :

**Comment by slevy85** [comment]
Yes it is better you are right. You can close it thank you very much
